### ISSUE TITLE
TextToFlowed doesn't generate valid flowed text

### ISSUE THREAD

**Issue by jessicah** [jstedfast/MimeKit#580]
**Describe the bug**
Using `TextToFlowed` to generate `format=flowed` output, it doesn't appear to generate valid output. Lines appear to be space-stuffed at the beginning.

**Platform (please complete the following information):**
 - OS: Windows
 - .NET Runtime: Core
 - .NET Framework: .NET Core 3.1
 - MimeKit Version: 2.6.0

**To Reproduce**
```
var flowedConverter = new MimeKit.Text.TextToFlowed();
var flowedText = flowedConverter.Convert(textWithParagraphsInSingleLines);
```

**Expected behavior**
Text should be correctly flowed with long lines broken up with proper line continuations.

**Comment by jessicah** [comment]
I think it might be this line: https://github.com/jstedfast/MimeKit/blob/master/MimeKit/Text/TextToFlowed.cs#L133

As it doesn't increment the index, the trailing space after splitting the line is processed again on subsequent calls to `GetFlowedLine`: https://github.com/jstedfast/MimeKit/blob/master/MimeKit/Text/TextToFlowed.cs#L130.

I believe the correct thing to do here is to increment index until a non-space character is encountered.

**Comment by jstedfast** [comment]
Do you happen to have a test case that I can use to see the issue and verify that it's fixed once it's fixed?

**Comment by jstedfast** [comment]
FWIW, so that we are all on the same page with regards to expected behavior, the specification for format=flowed is: https://tools.ietf.org/html/rfc3676

The test case I just added above (it's split between 2 commits because I can't get VS2019 to run the unit tests for some reason so had to commit a partial test in order to complete it on my MacBook w/ VS4Mac which runs them fine).

https://github.com/jstedfast/MimeKit/commit/b4a92ebd77f15581a351c8dfdd0eebf7edd978b0 adds the expected output which I added a comment to which exposes at least 1 bug but I don't think that's the bug that you are hitting.

Secondly, keep in mind that according to my docs for `TextToFlowed`, the output is designed for `format=flowed; delsp=yes` which has a different expected output than `delsp=no`.

Maybe `TextToFlowed` should be more configurable to dictate what `delsp` behavior to use.

I saw your fork & fix and I'm not sure that your fix is correct because because if you set `delsp=yes`, then your fix looks like it would break `FlowedToText` when `DeleteSpace=true`.
